### ISSUE TITLE
Why is `--cname` a required option

### ISSUE THREAD

**Issue by awentzel** [Azure/azure-cli#13557]
When it's already required in `az network dns record-set cname`? 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6bf2496b-338d-9d05-7c91-63b9ed576dae
* Version Independent ID: 0ae8e4a4-59ae-c756-9e5b-44239188882f
* Content: [az network dns record-set cname](https://docs.microsoft.com/en-us/cli/azure/network/dns/record-set/cname?view=azure-cli-latest#az-network-dns-record-set-cname-set-record)
* Content Source: [src/azure-cli/azure/cli/command_modules/network/_help.py](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/network/_help.py)
* Service: **virtual-network**
* GitHub Login: @rloutlaw
* Microsoft Alias: **routlaw**

**Comment by yungezz** [comment]
hi @MyronFanQiu could you pls have a look?

**Comment by yonzhan** [comment]
add to S170

**Comment by MyronFanQiu** [comment]
In `az network dns record-set cname create` command, `--cname` is not required. In `az network dns record-set cname set-record`, `--cname` is required. Could your share more about your usage scenario so that I can understand the context? Since we need to set a record, it makes sense to me that `--cname` is required.

**Comment by awentzel** [comment]
@MyronFanQiu I think the terminology is a little confusing for me, but, perhaps this is because I'm not a network expert.  I eventually found the right command configuration.  I think that biggest thing that's confusing is when Azure CLI documentation lists parameter options and refer to name, resource group, location, cname etc, but, often when it's linking for example a web app to a dns record entry it's not clear if those options are for the dns record or the web app.  It would be much more useful to explain this detail "--name of the web app resource" for example.  

A better example for this use case is `--cname` I've also used that as the key name, but, actually it's the value.  This isn't clear either.  `--cname some-site.azurewebsites.net` is really what's being asked for.  Maybe a better option name would be `--cname-alias` and `--record-set-value` for example.  both options that map directly to UI on the Azure Portal.

I'd recommend using both `--record-set-name` and `--record-set-value` and eliminate -cname.

That would seem more intuitive, IMHO.
